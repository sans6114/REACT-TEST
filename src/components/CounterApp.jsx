import {
  useEffect,
  useState,
} from 'react';

import PropTypes from 'prop-types';

import { Button } from './Button';

const CounterApp = ({ value, title }) => {
    const [counter, setCounter] = useState(value)
    useEffect(() => {
        console.log('useeffect!')
    }, [counter])

    const incrementFunction = () => {
        setCounter(counter + 1)
    }

    const decrementFunction = () => {
        if (counter > 0) {
            // setCounter(counter - 1)
            setCounter((c) => c - 1)
        }
    }


    const resetFunction = () => {
        setCounter(value)
    }

    return (
        <section className='h-screen flex flex-col items-center'>
            <h2 className='mb-4 text-2xl font-extrabold leading-none tracking-tight text-gray-900 dark:text-white text-center mt-10'>{title}</h2>
            <h2 className='font-extrabold text-8xl'> {counter} </h2>
            <div className='flex justify-between w-full mt-10'>
                <Button btnText={'Increment'} onClick={incrementFunction} />
                <Button btnText={'Reset'} onClick={resetFunction} />
                <Button btnText={'Decrement'} onClick={decrementFunction} />
            </div>
        </section>
    )
}

export default CounterApp

CounterApp.propTypes = {
    title: PropTypes.string,
    value: PropTypes.number.isRequired,
};