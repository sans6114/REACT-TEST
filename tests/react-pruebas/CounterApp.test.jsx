import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import CounterApp from '../../src/components/CounterApp';

describe('test on <CounterApp/>', () => {


    const title = 'Hola, Soy title';
    const value = 100

    test('debe hacer match con el snapshot', () => {

        const { container } = render(<CounterApp title={title} value={value} />);


        expect(container).toMatchSnapshot()
    });

    test('debe mostrar el valor inicial de 100', () => {
        // Renderizamos el componente con title y value
        const { getByText } = render(<CounterApp title={title} value={value} />);

        // Verificamos que el valor inicial de 100 estÃ© presente en el componente
        expect(getByText(value.toString())).toBeTruthy();
    });



    test('debe de retornar un evento click que incremente value en una unidad', () => {
        
        render(<CounterApp title={title} value={value} />);
        fireEvent.click(screen.getByText('Increment'))
        expect(screen.getByText('101')).toBeTruthy()


    });
    
    test('debe de retornar un evento click que decremente value en una unidad', () => {
        
        render(<CounterApp title={title} value={value} />);
        fireEvent.click(screen.getByText('Decrement'))
        expect(screen.getByText('99')).toBeTruthy()


    });

        
    test('debe de retornar un evento click que rectee value a su valor original', () => {
        
        render(<CounterApp title={title} value={value} />);
        fireEvent.click(screen.getByText('Decrement'))
        fireEvent.click(screen.getByText('Decrement'))
        expect(screen.getByText('98')).toBeTruthy()

        fireEvent.click(screen.getByText('Reset'))
        expect(screen.getByText('100')).toBeTruthy()
        screen.debug()


    });
});
