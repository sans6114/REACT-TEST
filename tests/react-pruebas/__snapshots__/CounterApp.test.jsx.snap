// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <CounterApp/> debe hacer match con el snapshot 1`] = `
<div>
  <section
    class="h-screen flex flex-col items-center"
  >
    <h2
      class="mb-4 text-2xl font-extrabold leading-none tracking-tight text-gray-900 dark:text-white text-center mt-10"
    >
      Hola, Soy title
    </h2>
    <h2
      class="font-extrabold text-8xl"
    >
       
      100
       
    </h2>
    <div
      class="flex justify-between w-full mt-10"
    >
      <button
        class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
        type="button"
      >
        Increment
      </button>
      <button
        class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
        type="button"
      >
        Reset
      </button>
      <button
        class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
        type="button"
      >
        Decrement
      </button>
    </div>
  </section>
</div>
`;
